pkgs/development/tools/database/schemaspy/default.nix

```diff
@@ -1,30 +1,83 @@
-{ lib, stdenv, fetchurl, jre, makeWrapper, graphviz }:
+{ lib
+, stdenv
+, callPackage
+, maven
+, jdk
+, jre
+, buildMaven
+, makeWrapper
+, git
+, fetchFromGitHub
+, graphviz
+, ensureNewerSourcesHook
+}:
 
-stdenv.mkDerivation rec {
-  version = "6.1.0";
+let
+  version = "6.1.1-SNAPSHOT";
   pname = "schemaspy";
 
-  src = fetchurl {
-    url = "https://github.com/schemaspy/schemaspy/releases/download/v${version}/${pname}-${version}.jar";
-    sha256 = "0lgz6b17hx9857fb2l03ggz8y3n8a37vrcsylif0gmkwj1v4qgl7";
+  src = fetchFromGitHub {
+    owner = "schemaspy";
+    repo = "schemaspy";
+    rev = "110b1614f9ae4aec0e4dc4e8f0e7c647274d3af6";
+    sha256 = "sha256-X5B34zGhD/NxcK8TQvwdk1NljGJ1HwfBp47ocbE4HiU=";
   };
 
-  dontUnpack = true;
+  deps = stdenv.mkDerivation {
+    name = "${pname}-${version}-deps";
+    inherit src;
+
+    nativeBuildInputs = [ jdk maven git ];
+    buildInputs = [ jre ];
+
+    buildPhase = ''
+      mvn package -Dmaven.test.skip=true -Dmaven.repo.local=$out/.m2 -Dmaven.wagon.rto=5000
+    '';
+
+    # keep only *.{pom,jar,sha1,nbm} and delete all ephemeral files with lastModified timestamps inside
+    installPhase = ''
+      find $out/.m2 -type f -regex '.+\(\.lastUpdated\|resolver-status\.properties\|_remote\.repositories\)' -delete
+      find $out/.m2 -type f -iname '*.pom' -exec sed -i -e 's/\r\+$//' {} \;
+    '';
+
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+    outputHash = "sha256-CUFA9L6qqjo3Jp5Yy1yCqbS9QAEb9PElys4ArPa9VhA=";
+
+    doCheck = false;
+  };
+in
+stdenv.mkDerivation rec {
+  inherit version pname src;
 
   buildInputs = [
-    jre
+    maven
   ];
 
   nativeBuildInputs = [
     makeWrapper
+    # the build system gets angry if it doesn't see git (even though it's not
+    # actually in a git repository)
+    git
+
+    # springframework boot gets angry about 1970 sources
+    # fix from https://github.com/nix-community/mavenix/issues/25
+    (ensureNewerSourcesHook { year = "1980"; })
   ];
 
   wrappedPath = lib.makeBinPath [
     graphviz
   ];
 
+  buildPhase = ''
+    VERSION=${version}
+    SEMVER_STR=${version}
+
+    mvn package --offline -Dmaven.test.skip=true -Dmaven.repo.local=$(cp -dpR ${deps}/.m2 ./ && chmod +w -R .m2 && pwd)/.m2
+  '';
```

Comment:
### nit
feel free to disregard 

---

```
-Dmaven.repo.local=$(cp -dpR ${deps}/.m2 ./ && chmod +w -R .m2 && pwd)/.m2
```
the copying of dependencies feels a little opaque, and I'm curious if there's a better way to represent this.

could the 
```
cp -dpR ${deps}/.m2 ./ && chmod +w -R .m2 && pwd)/.m2
```
 be done in a `configurePhase` or some other [phase](https://nixos.org/manual/nixpkgs/stable/#var-stdenv-phases)?

it'd also be good to do so that invoking `$buildPhase` repeatedly in a dev environment doesn't copy the static dependencies multiple times.

Maybe something like this would work?
```suggestion
  configurePhase = ''
    cp -dpR ${deps}/.m2 ./ &&
    chmod +w -R ./.m2
  '';

  buildPhase = ''
    VERSION=${version}
    SEMVER_STR=${version}

    mvn package --offline -Dmaven.test.skip=true -Dmaven.repo.local=$PWD/.m2
  '';
```